File: Java/src/main/java/Solution701_800/Solution704.java
package Solution701_800;

import com.sun.java.swing.plaf.windows.resources.windows;

/**
 * @author windows
 */
public class Solution704 {
    public int search(int[] nums, int target) {
        int left = - 1;
        int right = nums.length;
        while (left != right - 1){
            int mid = left + (right - left) / 2;
            if (nums[mid] == target){
                return mid;
            }else if (nums[mid] < target){
                left = mid;
            }else {
                right = mid;
            }
        }
        return -1;
    }


    public static void main(String[] args) {
        Solution704 solution704 = new Solution704();
    }
}

File: C++/src/Solution701_800/Solution704.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution704.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

int Solution704::search(vector<int> &nums, int target) {
    int left = 0, right = nums.size() - 1;
    while (left <= right) {
        int mid = (left + right) / 2;
        if (nums[mid] == target) {
            return mid;
        }else if (nums[mid] < target) {
            left = mid + 1;
        } else{
            right = mid - 1;
        }
    }
    return -1;
}
